### PR TITLE
fix(hex-grid): stop movement border misreading as geometry, ground wall frontier, bias camera to player

### PR DESCRIPTION
**Pull request by KirkDiggler**
Closes #456
Closes #457

Game Screen polish pair, diagnosed while investigating the "walls are broken" false alarm in the wave-1 closing playtest (see [PR #455's false-alarm investigation](https://github.com/KirkDiggler/rpg-dnd5e-web/pull/455) for the original trace).

## #456 — MovementRangeBorder read as broken geometry

Two compounding problems, both in how `MovementRangeBorder` decides what to draw:

1. **Outlined teammates as holes.** `isBlocked` (`HexGrid.tsx`) is entity-aware — it treats any hex occupied by another entity as unreachable, which is correct for real pathfinding (`useHexInteraction`) but was also being used to compute the border's reachable set (`useMovementRange`). An adjacent ally created a hex-shaped cyan "hole" in the boundary, reading as broken/special terrain rather than "you can't stop on your friend." Split off `isTerrainBlocked` (walls/floor only, no entity check) for the border's reachability calc — real pathfinding is untouched.
2. **Rendered always-on in FREE_ROAM.** `EncounterView.tsx`'s `isMyTurn` prop is forced `true` outside `TURN_BASED` so clicks still dispatch during exploration — but `HexGrid.tsx` was using that same flag to gate the border, so it was live and idle at spawn, not just during actionable moments. Added `shouldShowMovementBorder` (new pure function, unit-tested): shows only on the local player's own `TURN_BASED` turn, or in `FREE_ROAM` while actively hovering/planning a move (`hoveredHex`/`pathPreview` non-empty). Idle exploration shows nothing.

**Copilot follow-up (commit 19e1fe1)**: the original fix split reachability onto a terrain-only `isTerrainBlocked` for the border while real pathfinding (`useHexInteraction`) kept the entity-aware `isBlocked` — a hex only reachable by passing *through* an occupied hex would be marked reachable by the border but refused by real movement. Fixed by reverting the border's reachable-set calc to the same entity-aware `isBlocked` pathing uses (never lies relative to real movement), and moving the teammate-hole fix to a pure rendering rule instead: `calculateBoundaryEdges` (now exported) takes an `isOccupiedByOtherEntity` predicate and skips drawing a boundary edge toward a neighbor that's merely ally-occupied — the hex is still correctly excluded from the reachable set, only the misleading "hole" line is suppressed. `isTerrainBlocked` removed.

## #457 — walls read as floating blocks; camera framed the action out of shot

1. **Ungrounded frontier.** Per-viewer reveal is working as designed — a wall enters the revealed set the moment sight reaches it, everything beyond stays black — but the net visual is a solid wall block standing at the edge of revealed floor with nothing behind it, reading as floating off-map rather than a wall bounding a room. Added `frontierGroundHintHexes` (`dungeonMapGeometry.ts`, unit-tested) + `FrontierGroundHint.tsx`: a dim, low-opacity hex rendered at every hex immediately beyond a wall that isn't already revealed floor. **Kind-agnostic by construction** — it works off wall geometry (`from`/`to` line), never `WallKind`, so it extends automatically to `DOOR_CLOSED`/`DOOR_OPEN`/`WINDOW` once wave 2 folds doors into this `walls` array.
2. **Camera drifted from the player.** The camera's pan target recentered to the bbox of *all* revealed floor tiles on every `GeometryRevealed` event — as the revealed area grew, the player (and any nearby walls) drifted from center, sometimes toward the frame's edge. Camera panning now seeds from a one-time stable base target (no more snap-to-growing-bbox), and framing is driven by `focusTarget`, which now continuously follows the local player's *live position* rather than a one-shot "turn change" trigger that, on inspection, never actually fired: `currentEntityId` is always the local player (`EncounterMap.tsx`/`PlaytestMap.tsx` both pass `currentEntityId={myEntityId}`), never the combat-active entity despite the name, so the old memo (keyed only on that stable id) recomputed once at mount and never again.

## Tests

- `src/hooks/dungeonMapGeometry.test.ts`: 7 new tests for `frontierGroundHintHexes` (isolated single-cell wall → 6 hints, excludes already-revealed neighbors, doesn't hint a hex that's itself a wall, dedupes shared neighbors between adjacent walls, computes hints along a full multi-cell line not just endpoints, empty/malformed-wall edge cases).
- `src/components/hex-grid/useMovementRange.test.ts`: 5 tests for `shouldShowMovementBorder` (shows on own TURN_BASED turn, hides on others' turns even mid-plan, hides idle FREE_ROAM, shows FREE_ROAM while planning, FREE_ROAM's always-true `isPlayerTurn` doesn't leak into the gate) + 7 new tests from the Copilot follow-up: `reachableHexes`/`findPath` agree a hex behind an occupied bottleneck is unreachable (and agree it *is* reachable when the gate is clear), and `calculateBoundaryEdges` suppresses exactly the ally-occupied edge while still drawing genuine terrain boundaries.
- Full suite (clean `npx vitest run`, no filters): `607 passed` across `38` files. `npm run ci-check` green on every check (format/lint/typecheck/build/test), on every commit including the Copilot follow-up, both manual and pre-push-hook runs.

## Visual verification

All screenshots captured live against the dev stack (api on merged `main`, Envoy `:8080`, this branch's vite on `:3004`), same deterministic 20x20 room, same camera-pan recipe for matched before/after pairs. **I viewed every frame below before writing its description.**

### #456 — before/after

**Before** (`fix/gamescreen-polish~1`, old code): bob and alice spawned adjacent, `FREE_ROAM`, round 0, completely idle (no hover, no click yet).

456 before

I viewed this frame; it shows the cyan `MovementRangeBorder` fully visible while idle (the outer perimeter, top-left/bottom-right), and a distinct hex-shaped cyan outline drawn directly around alice — the exact bug: always-on, and a teammate treated as a "hole".

**After** (this branch), identical spawn/idle scenario:

456 after idle

I viewed this frame; it shows bob and alice standing adjacent in `FREE_ROAM` at idle — zero cyan border anywhere in the scene. No outline around alice, no idle always-on line.

**After**, hovering a hex to plan a move (same session):

456 after hover

I viewed this frame; it shows the path-preview highlight near bob (blue tiles), the cyan `MovementRangeBorder` now visible tracing the reachable-range perimeter (top-left/bottom-right corners), and alice standing nearby with no hole or outline around her — the border is actionable-only and still correctly excludes an ally from the boundary set even while showing.

**Re-verification after the Copilot semantics fix (commit 19e1fe1)**, same idle scenario:

459 refresh idle

I viewed this frame; it shows bob and alice adjacent at idle `FREE_ROAM` — still zero border, unchanged from before the semantics fix.

Same hover-to-plan scenario, re-verified:

459 refresh hover

I viewed this frame; it shows the same path-preview and border-perimeter behavior as before, alice still with no hole around her — visually identical to the pre-fix screenshot above, confirming the semantics fix changed only the underlying reachability correctness, not the rendered result for this scenario.

### #457 — before/after

**Before**, wall cluster at the reveal frontier, old code:

457 before frontier

I viewed this frame; it shows three solid wall clusters sitting in pure, sharp-edged black — no ground, no transition, blocks floating in the void.

**After**, same wall clusters, this branch:

457 after frontier wide

I viewed this frame; it shows the same three clusters, now each with a faint dim halo extending beyond its base rather than a hard cut to black.

Closer, single-cluster crop:

457 after frontier closeup

I viewed this frame; it shows solid wall blocks with visible top/side shading, each ringed by a subtly-lighter-than-pure-black hex halo (`FrontierGroundHint`) grounding it against the darkness beyond.

**Camera bias — before**: same 4 click-to-move steps from spawn, old code, final position:

457 before camera drift

I viewed this frame; it shows bob noticeably right-of-center after 4 moves (revealed area: 412 tiles) — the camera centered on the growing revealed-area bbox, not on him.

**Camera bias — after**: same 4 click-to-move steps from spawn, this branch:

457 after camera follows

I viewed this frame; it shows bob still essentially centered after 4 moves (revealed area: 490 tiles, larger than the before case) — the camera followed him instead of drifting toward the bbox center.

## Product-feel call

These are visual/feel changes — implemented my best take per the issues' suggested directions, verified functionally correct and visually distinct from the "before" state above. Deferring the actual feel judgment to Kirk at review time, as directed. Specific choices I made without a second option in the issue text: `FrontierGroundHint`'s dimness (`opacity: 0.35`, color `0x232323` — dim enough to never read as "real, walkable floor," visible enough to read as "there's more here"); `MovementRangeBorder`'s "planning a move" trigger keyed off existing `hoveredHex`/`pathPreview` state rather than new interaction tracking; camera follow uses the existing smooth-lerp mechanism (no new easing/mode toggle) rather than adding a follow-cam/overview-mode switch, since continuous player-following via the existing lerp already satisfied the issue's core complaint without new UI surface.

## Pushback

None on either brief.

### CODE REVIEW

File: src/components/hex-grid/HexGrid.tsx [hunk 1]
```diff
@@ -277,14 +333,38 @@ function Scene({
...
+  // Use movement range hook for boundary visualization. Terrain-only
+  // blocking (see isTerrainBlocked above) — allies don't punch holes in
+  // the reachable set.
   const { boundaryEdges } = useMovementRange({
     entityPosition: currentEntityPosition,
     movementRemaining,
     hexSize: HEX_SIZE,
-    isBlocked,
+    isBlocked: isTerrainBlocked,
   });
```
[the remark this answers is not in the document]
Comment (KirkDiggler) [reply src/components/hex-grid/HexGrid.tsx:350]: Valid, fixed in 19e1fe1. Reachability for the border now derives from the same entity-aware isBlocked real pathing uses, so the reachable set never lies. The teammate-hole issue (the reason isBlocked was split off in the first place) is instead solved as a rendering-only rule: calculateBoundaryEdges now takes an isOccupiedByOtherEntity predicate and skips drawing a boundary edge toward a neighbor that's only unreachable because an ally is standing on it — that hex is still correctly excluded from the reachable set. Added tests pinning both invariants: reachableHexes/findPath agree a hex behind an occupied bottleneck is unreachable, and calculateBoundaryEdges suppresses exactly the ally-occupied edge while still drawing genuine terrain boundaries.
